### DISCUSSION TITLE
Version consistency & postgrest local testing

### DISCUSSION THREAD

**Discussion by wiverson** [supabase-community/supabase-csharp#77] [General]
I'm working on a PR for https://github.com/supabase-community/postgrest-csharp, and I'm noticing that there appears to be some inconsistency in the version[s] of .NET used for the projects & builds. I'm also getting test failures that appear to be related to LINQ queries.

I *think* the best option would be to target the following, either [netstandard2.0 or netstandard2.1](https://learn.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-1#net-standard-not-deprecated).

```xml
<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <TargetFrameworks>netstandard2.1;net48</TargetFrameworks>
        <LangVersion>9.0</LangVersion>
    </PropertyGroup>
```

This happens to be the same levels of support for the current [LTS version of Unity](https://docs.unity3d.com/2022.3/Documentation/Manual/dotnetProfileSupport.html).

Right now it appears that some of the other projects (e.g. the [PostgrestTests](https://github.com/supabase-community/postgrest-csharp/blob/master/PostgrestTests/PostgrestTests.csproj)) are built targeting net7.0. If I leave it targeting net7.0  the build works but I'm getting 49 test failures out of 78 tests. Looks like all of the failures are related to LINQ queries.

Here's one example of test failure I'm getting. This is targeting net7.0, docker compose up seems to be working fine. macOS, Rider.

```
Test method PostgrestTests.ClientApi.TestColumns threw exception: 
Postgrest.RequestException: duplicate key value violates unique constraint "movie_pkey"
    at Postgrest.Helpers.MakeRequest(ClientOptions clientOptions, HttpMethod method, String url, JsonSerializerSettings serializerSettings, Object data, Dictionary`2 headers, CancellationToken cancellationToken) in /Users/will/src/supabase-community/postgrest-csharp/Postgrest/Helpers.cs:line 117
   at Postgrest.Helpers.MakeRequest[T](ClientOptions clientOptions, HttpMethod method, String url, JsonSerializerSettings serializerSettings, Object data, Dictionary`2 headers, Func`1 getHeaders, CancellationToken cancellationToken) in /Users/will/src/supabase-community/postgrest-csharp/Postgrest/Helpers.cs:line 45
   at PostgrestTests.ClientApi.TestColumns() in /Users/will/src/supabase-community/postgrest-csharp/PostgrestTests/ClientApi.cs:line 1125
```

I think the first step would be to sort out/confirm a working net7.0 build w/tests. Any thoughts on possible tweaks/fixes to get the LINQ-related stuff working?

The second step I think would be to be more explicit about the recommended target settings in all of the projects.

FWIW I'm using degit to drop source into Unity for subprojects right now (easier to debug). It's working great for auth, only thing I'm running into with postgrest-csharp is it's complaining about [file based namespaces are C# 10](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/file-scoped-namespaces) and Unity only goes up to C# 9. It's a minor tweak but it's making my local copy drift. It would be a very minor PR to swap them back to C# 9 namespaces, but I don't want to do a PR until I have working tests locally.

**Comment by acupofjose** [comment]
Good catch. They should all be targeting `netstandard2.0` - we can move to C# 9 namespaces, the change was one that Rider had suggested.

As for the failing tests, try doing a `docker-compose down -v` on that project. A recent release changed the database schema that should fix the issue for you!

**Comment by wiverson** [comment]
Hey you are up! (are you on Europe time? FWIW I'm in Seattle/PST).

Good news, I got a clean build/run in Rider. The `docker-compose down -v` seemed to do the trick for the tests. To get the project to build/run I had to set all three csproj `<TargetFramework>net7.0</TargetFramework>`, and also comment out `<Nullable>enable</Nullable>`.

Would you like me to go ahead and send in a PR to move all three postgrest projects back to netstandard2.0 as a target and lang level 9? Off the cuff I think the only thing missing was a few lines to polyfill back HttpUtility query parse, I think maybe a HashCode polyfill as well.

**Reply by acupofjose** [reply]
Up indeed! Just odd hours of computer access - I'm Chicago/CST. 

And yes, a PR back to net2.0 would be marvelous . Thanks @wiverson!
